lib/core/base/virtual-node.js

```diff
@@ -1,15 +1,42 @@
 const whitespaceRegex = /[\t\r\n\f]/g;
 
+class AbstractVirtualNode {
+	constructor() {
+		this.children = [];
```

Comment:
No big deal (no need to change), but this could just be:

```
class AbstractVirtualNode {
  children = []
  parent = null

  // ...
}
```

Reply:
I forget we have babel to correct that for us an it's not just a native class